drivers/regulator/regulator_max20335.c

```diff
@@ -0,0 +1,383 @@
+	case MAX20335_PMIC_SOURCE_LDO3:
+		return regulator_max20335_set_buck_ldo_voltage(dev, min_uv, max_uv, &ldo3_range,
+							       MAX20335_LDO3_VSET);
+	default:
+		return -ENOTSUP;
+	}
+}
+
+static unsigned int regulator_max20335_count_current_limits(const struct device *dev)
+{
+	const struct regulator_max20335_config *config = dev->config;
+
+	switch (config->source) {
+	case MAX20335_PMIC_SOURCE_BUCK1:
+		return linear_range_values_count(&buck1_current_limit_range);
+	case MAX20335_PMIC_SOURCE_BUCK2:
+		return linear_range_values_count(&buck2_current_limit_range);
+	default:
+		return -ENOTSUP;
+	}
+}
+
+static int regulator_max20335_list_current_limit(const struct device *dev, unsigned int idx,
+						 int32_t *current_ua)
+{
+	const struct regulator_max20335_config *config = dev->config;
+
+	switch (config->source) {
+	case MAX20335_PMIC_SOURCE_BUCK1:
+		return linear_range_get_value(&buck1_current_limit_range, idx, current_ua);
+	case MAX20335_PMIC_SOURCE_BUCK2:
+		return linear_range_get_value(&buck2_current_limit_range, idx, current_ua);
+	default:
+		return -ENOTSUP;
+	}
+}
+
+static int regulator_max20335_set_current_limit(const struct device *dev,
+						int32_t min_ua,
+						int32_t max_ua)
+{
+	const struct regulator_max20335_config *config = dev->config;
+	uint8_t val;
+	uint16_t idx;
+	int ret;
+
+	ret = i2c_reg_read_byte_dt(&config->bus, MAX20335_BUCK12_CSET, &val);
+	if (ret < 0) {
+		return ret;
+	}
+
+	switch (config->source) {
+	case MAX20335_PMIC_SOURCE_BUCK1:
+		ret = linear_range_get_win_index(&buck1_current_limit_range, min_ua, max_ua, &idx);
+		if (ret == -EINVAL) {
+			return ret;
+		}
+
+		val = idx | (val & MAX20335_BUCK1_CSET_MASK);
+		break;
+	case MAX20335_PMIC_SOURCE_BUCK2:
+		ret = linear_range_get_win_index(&buck2_current_limit_range, min_ua, max_ua, &idx);
+		if (ret == -EINVAL) {
+			return ret;
+		}
+
+		val = (idx << MAX20335_BUCK2_CSET_SHIFT) | (val & MAX20335_BUCK2_CSET_MASK);
+		break;
+	default:
+		return -ENOTSUP;
+	}
+
+	return i2c_reg_write_byte_dt(&config->bus, MAX20335_BUCK12_CSET, val);
+}
+
+static int regulator_max20335_init(const struct device *dev)
+{
+	const struct regulator_max20335_config *config = dev->config;
+
+	if (!i2c_is_ready_dt(&config->bus)) {
+		return -ENODEV;
+	}
+
+	regulator_common_data_init(dev);
+
+	return regulator_common_init(dev, false);
+}
+
+static const struct regulator_driver_api api = {
+	.enable = regulator_max20335_enable,
+	.disable = regulator_max20335_disable,
+	.set_mode = regulator_max20335_set_mode,
+	.count_voltages = regulator_max20335_count_voltages,
+	.list_voltage = regulator_max20335_list_voltage,
+	.set_voltage = regulator_max20335_set_voltage,
+	.get_voltage = regulator_max20335_get_voltage,
+	.count_current_limits = regulator_max20335_count_current_limits,
+	.list_current_limit = regulator_max20335_list_current_limit,
+	.set_current_limit = regulator_max20335_set_current_limit,
+};
+
+#define REGULATOR_MAX20335_DEFINE(node_id, id, _source)			\
```

Comment:
Whitespace looks a little odd for the define, did you do this manually or use clang-format?

Reply:
It was done by emacs, it should be ok now.